### ISSUE TITLE
Training dataset for HMR2A model

### ISSUE THREAD

**Issue by forseeking** [shubham-goel/4D-Humans#173]
First of all, thank you for releasing the HMR2 codebase and the pre-trained models.
While trying to reproduce the HMR2A training， I ran into some confusion about the datasets.

According to configs of hmr2a checkpoint,  the training loop mixes four datasets

TRAIN:
  H36M-TRAIN:        WEIGHT: 0.3
  MPII-TRAIN:        WEIGHT: 0.1
  COCO-TRAIN-2014:   WEIGHT: 0.4
  MPI-INF-TRAIN:     WEIGHT: 0.2
However, the training datasets provided for download seems only for hmr2b model, which have slightly different names / suffixes:

COCO-TRAIN-2014-WMASK-PRUNED
MPII-TRAIN-WMASK
MPI-INF-TRAIN-PRUNED
H36M-TRAIN-WMASK

Suffixes
• WMASK – segmentation / visibility mask?
• PRUNED – images filtered out? any other preprocessing?

Config mapping
Are COCO-TRAIN-2014-WMASK-PRUNED, MPII-TRAIN-WMASK, etc. the exact sets referred to as COCO-TRAIN-2014, MPII-TRAIN … in hmr2a.yaml, or do the plain names point to different (larger) versions?

Could you share (or confirm) the dataset packages actually used for the HMR2A results—download links / size / checksum—so we can train from scratch ？
Best regards,
Edwin

**Comment by CyberChickZ** [comment]
> First of all, thank you for releasing the HMR2 codebase and the pre-trained models. While trying to reproduce the HMR2A training， I ran into some confusion about the datasets.
> 
> According to configs of hmr2a checkpoint, the training loop mixes four datasets
> 
> TRAIN: H36M-TRAIN: WEIGHT: 0.3 MPII-TRAIN: WEIGHT: 0.1 COCO-TRAIN-2014: WEIGHT: 0.4 MPI-INF-TRAIN: WEIGHT: 0.2 However, the training datasets provided for download seems only for hmr2b model, which have slightly different names / suffixes:
> 
> COCO-TRAIN-2014-WMASK-PRUNED MPII-TRAIN-WMASK MPI-INF-TRAIN-PRUNED H36M-TRAIN-WMASK
> 
> Suffixes • WMASK – segmentation / visibility mask? • PRUNED – images filtered out? any other preprocessing?
> 
> Config mapping Are COCO-TRAIN-2014-WMASK-PRUNED, MPII-TRAIN-WMASK, etc. the exact sets referred to as COCO-TRAIN-2014, MPII-TRAIN … in hmr2a.yaml, or do the plain names point to different (larger) versions?
> 
> Could you share (or confirm) the dataset packages actually used for the HMR2A results—download links / size / checksum—so we can train from scratch ？ Best regards, Edwin


Hi, I have the same confusion regarding the datasets mentioned for HMR2A training.

Also, may I ask how you solved the **H36M evaluation issue**?
I used the Human3.6M dataset downloaded from [[Hugging Face](https://huggingface.co/)](https://huggingface.co) and [[this Google Drive link](https://drive.google.com/drive/folders/1r0B9I3XxIIW_jsXjYinDpL6NFcxTZart)](https://drive.google.com/drive/folders/1r0B9I3XxIIW_jsXjYinDpL6NFcxTZart), but the file names don’t match the ones referenced in the training scripts (from `fetch_training_data.sh`).

My current evaluation results for **HMR2.0a** are:

* `3DPW-TEST, mode_re: 44.5093`
* `3DPW-TEST, mode_mpjpe: 69.9`
* `COCO-VAL, mode_kpl2: 0.0009477`
* `COCO-VAL, kp0_pck_0.05: 0.950`

These seem within a reasonable range, but

* `H36M-VAL-P2, mode_re: 170.46`
* `H36M-VAL-P2, mode_mpjpe: 275.52`
  are **far beyond** expected values.

It looks like the decoded data from `fetch_training_data.sh` doesn’t fully align with the expected structure and resolution.
Could you please share how you handled the **H36M evaluation** part and which dataset version you used?

Thanks in advance!

Best,
Haochuan
